VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimLettersScreen/ClaimLettersScreen.tsx

```diff
@@ -36,29 +39,38 @@ const ClaimLettersScreen = ({ navigation }: ClaimLettersScreenProps) => {
   const { t } = useTranslation(NAMESPACE.COMMON)
   const theme = useTheme()
   const dispatch = useAppDispatch()
-  const { loading, decisionLetters, downloading } = useSelector<RootState, DecisionLettersState>(
-    (state) => state.decisionLetters,
-  )
+  const queryClient = useQueryClient()
   const claimsInDowntime = useDowntime(DowntimeFeatureTypeConstants.claims)
   const prevScreen = useNavigationState((state) => state.routes[state.routes.length - 2]?.name)
-
+  const { demoMode } = useSelector<RootState, DemoState>((state) => state.demo)
+  const [letterID, setLetterID] = useState<string>('')
+  const {
+    data: decisionLettersData,
+    isLoading: loading,
+    refetch: fetchInfoAgain,
+  } = useDecisionLetters({
+    enabled: screenContentAllowed('WG_ClaimLettersScreen') && !claimsInDowntime,
+  })
+  const {
+    isLoading: downloading,
+    error: downloadLetterErrorDetails,
+    refetch: refetchLetter,
+  } = useDownloadDecisionLetter(letterID, demoMode, {
+    enabled: letterID.length > 0,
+  })
   // This screen is reachable from two different screens, so adjust back button label
+  const decisionLetters = decisionLettersData?.data || ([] as DecisionLettersList)
   const backLabel = prevScreen === 'ClaimDetailsScreen' ? t('claimDetails.title') : t('claims.title')
 
-  const snackbarMessages: SnackbarMessages = {
-    successMsg: '',
-    errorMsg: t('claimLetters.download.error'),
-  }
-
   useEffect(() => {
-    if (screenContentAllowed('WG_ClaimLettersScreen') && !claimsInDowntime) {
-      dispatch(getDecisionLetters(ScreenIDTypesConstants.DECISION_LETTERS_LIST_SCREEN_ID))
+    if (downloadLetterErrorDetails && isErrorObject(downloadLetterErrorDetails)) {
+      if (!snackBar) {
+        logAnalyticsEvent(Events.vama_snackbar_null('ClaimLetters view letter'))
+      }
+      snackBar?.hideAll()
+      showSnackBar(t('claimLetters.download.error'), dispatch, fetchInfoAgain, false, true, true)
```

Comment:
Shouldn't the retry function be `refetchLetter` instead, since this is dependent on `downloadLetterErrorDetails`?

Reply:
you are right, I updated it